### PR TITLE
FEM-1418 - Kaltura stats integrarton/fixes

### PR DESCRIPTION
**Pull request by giladna**

### CODE REVIEW

File: playkit/src/main/java/com/kaltura/playkit/plugins/KalturaStatsPlugin.java [hunk 1]
```diff
@@ -292,6 +302,13 @@ public void onEvent(PKEvent event) {
                             durationValid = true;
                         }
                         break;
+                    case ENDED:
```
Comment (AntonAFA) [inline playkit/src/main/java/com/kaltura/playkit/plugins/KalturaStatsPlugin.java]: Why send it at all this events here?

File: playkit/src/main/java/com/kaltura/playkit/player/PlayerController.java [hunk 2]
```diff
@@ -93,8 +93,13 @@ public void onEvent(PlayerEvent.Type eventType) {
                 // TODO: use specific event class
                 switch (eventType) {
                     case DURATION_CHANGE:
-                        event = new PlayerEvent.DurationChanged(getDuration());
-                        if (getDuration() != Consts.TIME_UNSET && isNewEntry) {
+                        long newDuration = getDuration();
+                        if (newDuration == Consts.TIME_UNSET) {
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/player/PlayerController.java]: Masking this event requires careful testing, someone could rely on it. Also must be compared to iOS.

File: playkit/src/main/java/com/kaltura/playkit/plugins/KalturaStatsPlugin.java [hunk 3]
```diff
@@ -63,30 +85,30 @@
         PLAY_REACHED_50(5),
         PLAY_REACHED_75(6),
         PLAY_REACHED_100(7),
-        OPEN_EDIT(8),
-        OPEN_VIRAL(9),
-        OPEN_DOWNLOAD(10),
-        OPEN_REPORT(11),
+        //OPEN_EDIT(8),
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/plugins/KalturaStatsPlugin.java]: Just delete the irrelevant events -- don't comment-out.

File: playkit/src/main/java/com/kaltura/playkit/plugins/KalturaStatsPlugin.java [hunk 4]
```diff
@@ -179,6 +183,12 @@ protected void onUpdateMedia(PKMediaConfig mediaConfig) {
         } else {
             baseUrl = DEFAULT_BASE_URL;
         }
+        if (Utils.isJsonObjectValueValid(pluginConfig, "timerInterval")) {
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/plugins/KalturaStatsPlugin.java]: Maybe it's a good time to create a Config class for this plugin. Then the app knows what to set.

File: playkit/src/main/java/com/kaltura/playkit/plugins/KalturaStatsPlugin.java [hunk 5]
```diff
@@ -179,6 +183,12 @@ protected void onUpdateMedia(PKMediaConfig mediaConfig) {
...
+            timerInterval = pluginConfig.getAsJsonPrimitive("timerInterval").getAsInt() * (int)Consts.MILLISECONDS_MULTIPLIER;
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/plugins/KalturaStatsPlugin.java]: In that new Config class, the `timerInterval` field will be called `timerIntervalMillis`. Let the application translate.

File: playkit/src/main/java/com/kaltura/playkit/player/PlayerController.java (hunk 2 above)
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/player/PlayerController.java]: We decided that negative duration should not be fired
Yes we will test

File: playkit/src/main/java/com/kaltura/playkit/plugins/KalturaStatsPlugin.java (hunk 3 above)
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/plugins/KalturaStatsPlugin.java]: done

File: playkit/src/main/java/com/kaltura/playkit/plugins/KalturaStatsPlugin.java (hunk 4 above)
Comment (giladna) [inline playkit/src/main/java/com/kaltura/playkit/plugins/KalturaStatsPlugin.java]: will do it for next version

**Pull request merged**